exporter/exporter.go

```diff
@@ -57,8 +59,106 @@ resource "lightstep_metric_dashboard" "exported_dashboard" {
 }
 `
 
-func escapeSpanQuery(input string) string {
-	return strings.Replace(input, "\"", "\\\"", -1)
+const unifiedDashboardTemplate = `
+resource "lightstep_dashboard" "exported_dashboard" {
+  project_name = var.project
+  dashboard_name = "{{.Attributes.Name}}"
+{{range .Attributes.Charts}}
+  chart {
+    name = "{{.Title}}"
+    rank = "{{.Rank}}"
+    type = "{{.ChartType}}"
+{{range .MetricQueries}}
+    query {
+      query_name          = "{{.Name}}"
+      display             = "{{.Display}}"
+      hidden              = {{.Hidden}}
+      query_string        = {{escapeQueryString .TQLQuery}}
+	}
+{{end}}
+  }
+{{end}}
+}
+`
+
+func escapeHCLString(input string) string {
+	input = strings.Replace(input, "\"", "\\\"", -1)
+	input = strings.Replace(input, "\n", "\\\t", -1)
+	input = strings.Replace(input, "\r", "\\\t", -1)
+	input = strings.Replace(input, "\t", "\\\t", -1)
+	input = strings.Replace(input, "\\", "\\\\", -1)
+	return input
+}
+
+func escapeQueryString(input string) string {
+	// Use "heredoc" syntax if the query contains any newlines or other characters that'd
+	// need to be escaped and make the single line representation less convenient to work with.
+	if strings.Contains(input, "\"") ||
+		strings.Contains(input, "\n") ||
+		strings.Contains(input, "\r") ||
+		strings.Contains(input, "\t") ||
+		strings.Contains(input, "\\") {
+		return "<<EOT\n" + input + "\nEOT"
+	} else {
+		return escapeHCLString(input)
```

Comment:
given that we're checking for all the same things, don't we know that `escapeHCLString` won't actually change `input`? would it be clearer to just return `input`?

Reply:
Good catch